javatests/arcs/android/systemhealth/testapp/StorageCore.kt

```diff
@@ -545,6 +584,36 @@ class StorageCore(val context: Context, val lifecycle: Lifecycle) {
         }
     }
 
+    @Suppress("UNCHECKED_CAST")
+    private suspend fun cleanerTask(
+        taskHandle: TaskHandle,
+        taskController: TaskController,
+        settings: Settings
+    ) = (taskHandle.handle as? ReadWriteCollectionHandle<TestEntity>)?.let { handle ->
+        val entities = List(settings.clearedEntities) {
+            SystemHealthTestEntity(settings.dataSizeInBytes)
+        }
+
+        if (entities.isNotEmpty()) {
+            withContext(handle.dispatcher) {
+                entities.map { handle.store(it) }
+            }.joinAll()
+            handle.getProxy().waitForIdle()
+        }
+
+        val elapsedTime = measureTimeMillis {
+            withContext(handle.dispatcher) { handle.clear() }.join()
```

Comment:
is joining appropriate here? (and in writerTask as well)
what does join here waits for? for the op to get to storage service?
I think clients of the handle class can use it without joining and it still should work fine

Reply:
Yes, we join for store, clear, client apis to wait for the op gets sent to outgoing channel and handled at ServiceStore's onProxyMessage. I did this by also referring to our codebase i.e. `dispatchStore` at HandleHelper.kt.

Reply:
note that dispatchStore is a testutil, not to be used in real code. But it's true that we join() in many places, we should try remove those at some point. in the meantime, maybe drop a comment here to explain why we join

Reply:
done